adapters/lora-vertx/src/main/java/org/eclipse/hono/adapter/lora/providers/JsonBasedLoraProvider.java

```diff
@@ -52,6 +64,42 @@ public LoraMessage getMessage(final RoutingContext ctx) {
         }
     }
 
+    String getFormattedUri(final String uriFormat,
+                           final String deviceId) {
+        return uriFormat.replaceAll("\\{\\{deviceId\\}\\}", deviceId);
+    }
+
+    @Override
+    public LoraCommand getCommand(final CommandEndpoint commandEndpoint, final String deviceId, final Buffer payload) {
+        Objects.requireNonNull(commandEndpoint);
+        Objects.requireNonNull(deviceId);
+        Objects.requireNonNull(payload);
+        Objects.requireNonNull(commandEndpoint.getUriFormat());
+        final URI uri;
+        try {
+            uri = new URI(getFormattedUri(commandEndpoint.getUriFormat(),
+                deviceId));
+        } catch (URISyntaxException e) {
+            LOG.error("Could not retrieve a valid uri for this commandEndpoint", e);
+            throw new IllegalStateException("Could not retrieve a valid uri for this commandEndpoint");
+        }
```

Comment:
I've updated this according to your remark. However, the fact that the `{{deviceId}}` placeholders can be replace with the actual deviceId made me do a replacement in the check as well. Otherwise this isn't a valid URI. I don't know if you agree with the strategy I'm following now...

Reply:
I think it's ok. 
Will the `{{deviceId}}` placeholder usually be used in the path or the query portion of the URL?
If it's in the query portion, we might want to URL-encode the device ID before doing the replacement, because the device ID may contain a ":" which is supposed to be encoded there. The other possible device ID characters should be ok (at least if the standard `RegistrationManagementConstants.DEFAULT_DEVICE_ID_PATTERN` is used).

Reply:
The example I had, was in the path itself and not in a query param. If this changes for a particular LNS, we should tackle that then, no?

Reply:
Yes, ok.